cmd2/cmd2.py

```diff
@@ -467,13 +467,17 @@ def perror(msg: Any, *, end: str = '\n', apply_style: bool = True) -> None:
             final_msg = "{}".format(msg)
         ansi.ansi_aware_write(sys.stderr, final_msg + end)
 
-    def pwarning(self, msg: Any, *, end: str = '\n') -> None:
-        """Apply the warning style to a message and print it to sys.stderr
+    def pwarning(self, msg: Any, *, end: str = '\n', apply_style: bool = True) -> None:
+        """Like perror, but applies ansi.style_warning by default
 
         :param msg: message to print (anything convertible to a str with '{}'.format() is OK)
         :param end: string appended after the end of the message, default a newline
+        :param apply_style: If True, then ansi.style_warning will be applied to the message text. Set to False in cases
+                            where the message text already has the desired style. Defaults to True.
         """
-        self.perror(ansi.style_warning(msg), end=end, apply_style=False)
+        if apply_style:
+            msg = ansi.style_warning(msg)
+        self.perror(msg, end=end, apply_style=False)
```

Comment:
This looks fine, but it makes me wonder if it would be better to refactor `perror` to accept an additional optional keyword-only argument called something like `style` and then have `pwarning` call `perror` and pass on `apply_style` and set `style = ansi.syle_warning(msg)`?